### DISCUSSION TITLE
How to join tables on multiple columns, without a "join table"?

### DISCUSSION THREAD

**Discussion by RonBarkan** [micronaut-projects/micronaut-data#2381] [Q&A]
I am trying to do a one-to-many join on multiple, non-pk columns, without much success.
I have tried using a unidirectional @OneToMany + @JoinColumn annotations, see [this example](https://github.com/RonBarkan/doublejoinbug/blob/master/src/test/java/com/example/JoinTest.java).

Looks like Micronaut data thinks it should use a join table "e_f", though I am looking at simply doing a left join on non primary keys.

I see this is the generated query:
```
SELECT e_."id",e_."cat_id",e_."name",e_."e_id",f_."id" AS f_id,f_."e_id" AS f_e_id,f_."cat_id" AS f_cat_id,f_."name" AS f_name
FROM "my_schema"."e" e_
LEFT JOIN "e_f" f_e_f_ ON e_."id"=f_e_f_."e_id" 
LEFT JOIN "my_schema"."f" f_ ON f_e_f_."f_id"=f_."id"
WHERE (e_."cat_id" = ? AND (e_.cat_id =?))
```

Note Micronaut ignored my join columns altogether.

How do I get it to do something like this instead?
```
SELECT
  e.id, e.cat_id, e.name, e.e_id,
  f.id, f.e_id, f.cat_id, f.name
FROM my_schema.e e
LEFT JOIN my_schema.f ON e.cat_id=f.cat_id AND e.e_id=f.e_id
WHERE e.cat_id=?
```

**Comment by iEAmi** [comment]
This is elso my problem.
